### ISSUE TITLE
Robustness checking of sklearn and catboost/xgboost models

### ISSUE THREAD

**Issue by val-co** [laudv/veritas#2]
Hello, thanks for sharing the code. I was wondering if you could help me understand how to check the robustness of a sklearn random forest or a catboost/xgboost trained model for some \epsilon (perturbation margin). Does the robustness checking function have the option for timeouts? 

Does this function also have an option for counterexample generation?

**Comment by lorenzocascioli** [comment]
Hello, you can look into the new notebook [examples/synthetic.ipynb](https://github.com/laudv/veritas/blob/main/examples/synthetic.ipynb): it shows how to execute a robustness search on a toy dataset, and it should contain all the answers to your questions.

Feel free to ask for further clarifications is something is not clear!

**Comment by val-co** [comment]
Thank you!

**Comment by val-co** [comment]
There's no support for Gradient Boosted Machines trained using Catboost?

**Comment by laudv** [comment]
Hi, there is currently no support for CatBoost, but it should not be too difficult to add. You can see the conversion functions for xgb [here](https://github.com/laudv/veritas/blob/main/src/python/veritas/xgb.py) and for scikit-learn trees [here](https://github.com/laudv/veritas/blob/main/src/python/veritas/sklearn.py). The conversion functions translate the xgboost/sklearn tree representatios into the internal Veritas AddTree representation.

The main challenge is to make sure that the trees are logically the same, especially when special split types (e.g. splits on categorical features) are used. Veritas currently only considers less than splits on single precision floats.

I unfortunately currently do not have time now to implement this, but I'm certainly willing to help and very happy to take a pull request.

**Comment by val-co** [comment]
Okay, thank you!
